docs/check_spelling.sh

```diff
@@ -0,0 +1,185 @@
+        cleanup 0
+    fi
+
+    if [ -z "$misspelled_wordlist" ]; then
+        printf "%s\n" "No typos found."
+        cleanup 0
+    else
+        if [ "$mode" == "all" ]; then
+            add_words_dict
+        fi
+        cleanup 1
+    fi
+}
+
+# Adds misspelled words to the custom dictionary
+function add_words_dict() {
+
+    if [ ! -z "$misspelled_wordlist" ]; then
+
+        printf "%s\n" "
+Add misspelled words into your custom dictionary or ignore the findings ?
+    * a[ll]     (add all words to the dictionary)
+    * o[ne]     (add words one by one to the dictionary)
+    * n[one]    (don't add, abort)"
+
+        while true; do
+            exec < /dev/tty
+            read answer
+            shopt -s nocasematch
+            case "$answer" in
+                a|all)
+                    add_words_all
+                    break ;;
+                o|one)
+                    add_words_manual
+                    break ;;
+                n|none)
+                    break ;;
+                *) 
+                    printf "%s\n" "Incorrect answer. Try again."
+                    continue
+            esac
+            shopt -u nocasematch
+        done
+    fi
+}
+
+# Adds all words to the custom dictionary
+function add_words_all() {
+
+    for word in $words; do
+        echo $word >> "$project_dict"
+        printf "%s\n" "\"$word\" added to the dictionary."
+    done
+}
+
+# Adds user selected words to the dictionary
+function add_words_manual() {
+
+    for word in $words; do
+        printf "%s\n" "Add this word to the dictionary: $word  (y[es] or n[o])"
+        while true; do
+            exec < /dev/tty
+            read answer
+            shopt -s nocasematch
+            case "$answer" in
+                y|yes)
+                    echo $word >> "$project_dict"
+                    printf "%s\n" "\"$word\" added to the dictionary."
+                    break ;;
+                n|no)
+                    break ;;
+                *) 
+                    printf "%s\n" "Incorrect answer, try again."
+                    continue
+            esac
+            shopt -u nocasematch
+        done
+    done
+}
+
+function cleanup() {
+
+    printf "%s\n" "Spell Check completed with exit code $1"
+    exit $1
+
+}
+
+function usage {
+    echo "usage: $0 [options]"\
+    echo "[options] is any of the following:"
+    echo -e "  -o \tdefault mode, lists typos one by one with suggestions"   
+    echo -e "  -a \tlists summary of all typos with possibility to add all,"
+    echo -e "     \texit code > 0 means there are spelling mistakes"
+    echo -e "  -q \tquiet mode, checks for spelling errors and exit"
+    echo -e "  -h \tshow this help"
+
+    exit 1
+}
+
+options=':oaqh'
+while getopts $options option
+do
+    case $option in
+        o  )    mode="one";;
+        a  )    mode="all";;
+        q  )    mode="quiet";;
+        h  )    usage;;
+    esac
+done
+
+shift $(($OPTIND - 1))
+
+
+add_additional_dicts
```

Comment:
Some short comments on why things are run in a certain order would be good here.

Reply:
Added.